### DISCUSSION TITLE
Do I need to include generated files on dart package?

### DISCUSSION THREAD

**Discussion by KeidsID** [rrousselGit/freezed#994] [Q&A] [answered]
I want seperate my freezed models into a dart package. Can I use the package if the generated files are git ignored?

It impossible right, or do build_runner also running on external package?

**Comment by SunlightBro** [comment, marked as answer]
pub.dev pulls packages as is in your .pub-cache directory, and marks for non-editing.

If you cmd-click into the code of one of your dependencies and try to edit the source, it should give you a warning.

**If your package relies on generated code, that code must be published.**

**Reply by KeidsID** [reply]
I see, thx for the clarity.

**Reply by SunlightBro** [reply]
To be honest I have not tested it, so I did not put it in the original answer.
But there is a file called `.pubignore` which will be used/applied when you run `dart pub publish`.

A common use case of it can be found inside the [freezed package](https://github.com/rrousselGit/freezed/tree/master/packages/freezed)

Remi uses a pubspec_overrides.yaml to override some dependencies during local development, but the file will get ignored when freezed gets published.

Now to the maybe (because I have not tried it yet), but if the syntax for .pubignore works like .gitignore then you could theoretically use `!` syntax in .pubignore to publish files that you normally have ignored in your git repo.

https://git-scm.com/docs/gitignore#_pattern_format
